docs/languages/powershell.md

```diff
@@ -0,0 +1,132 @@
+
+![Open PowerShell Examples](./images/powershell/pwshExamples.png)
+
+## The PowerShell Extension includes the following major features
+
+* Syntax Highlighting
+* Code Snippets
+* Intellisense for cmdlets and more
+* Rule-Based analysis provided by PowerShell Script Analyzer
+* Go to definition of cmdlets and variables
+* Find references of cmdlets and variables
+* Document and Workspace symbol discovery
+* Run selected selection of PowerShell code using F8
+* Launch online help for the symbol under the cursor using ctrl+F1
+* Local script debugging and basic interactive console support
+
+## Plaster
+
+ Plaster is a template-based file and project generator written in PowerShell. It's Purpose is to streamline the creation of PowerShell module projects, Pester tests, DSC configurations and more.
+ See the [Github Plaster repo](https://github.com/PowerShell/Plaster) for more information.
+
+ [Cmdlet Documentation](https://github.com/PowerShell/Plaster/blob/master/docs/en-US/Plaster.md)
+
+## Debugging
+
+![HSG](./images/powershell/HSGavatar.png)
+
+For more information on debugging check out the Hey, Scripting Guy two-part blog post series Written by [@keithHill](https://twitter.com/r_keith_hill) on debugging with the PowerShell Extension:
+
+[Debugging PowerShell script in Visual Studio Code – Part 1](https://blogs.technet.microsoft.com/heyscriptingguy/2017/02/06/debugging-powershell-script-in-visual-studio-code-part-1/)
+
+[Debugging PowerShell script in Visual Studio Code – Part 2](https://blogs.technet.microsoft.com/heyscriptingguy/2017/02/13/debugging-powershell-script-in-visual-studio-code-part-2/)
+
+## Settings Reference
+
+You can customize VSCode settings from the file menu by selecting preferences then clicking settings.
+
+You can also click the gear Icon
+located in the lower left corner of the activity bar.
+
+![codeGear](./images/powershell/codeGear.png)
+
+You can also use the shortcut <kbd>ctrl</kbd>-<kbd>,</kbd> (<kbd>cmd</kbd>-<kbd>,</kbd>) for Mac for a quick search
+
+This will launch the settings.json file, in VSCode versions prior to 1.27.1. , where the VSCode team has introduced a settings GUI, which will be the new default interface for customizing settings.
+
+You can still launch the JSON settings file by using the command __Open Settings (JSON)__ or by changing the default settings editor with the ```"workbench.settings.editor"```
+
+### VSCode Settings GUI
+
+![VSCode Setting GUI](./images/powershell/vscodeSettingsGui.png)
+
+### PowerShell Extension Settings
+
+![PowerShell Extension VSCode Settings](./images/powershell/pwshCodeSettings.png)
+
+## PowerShell Extension multi-version support
+
+You can configure the PowerShell Extensions to use any version of PowerShell installed on your machine by following [these instructions](https://docs.microsoft.com/en-us/powershell/scripting/core-powershell/vscode/using-vscode?view=powershell-6#using-a-specific-installed-version-of-powershell)
+
+## Pester
```

Comment:
Pester also has a cool integration in the VSCode extension:

![image](https://user-images.githubusercontent.com/2644648/45564557-a8bb6200-b805-11e8-8413-3ef1045afeb9.png)

It gives you CodeLens for `Run tests` and `Debug tests` (ignore the others)